EventFilter/CTPPSRawToDigi/src/RPixErrorChecker.cc

```diff
@@ -73,57 +93,111 @@ bool RPixErrorChecker::checkTrailer(bool& errorsInEvent, int fedId, unsigned int
   if ( fedTrailer.fragmentLength()!= nWords) {
     LogError("FedTrailerLenght")<< "fedTrailer.fragmentLength()!= nWords !! Fed: " << fedId << ", errorType = 34";
     errorsInEvent = true;
- 
+    if(includeErrors) {
+      int errorType = 34;
+      CTPPSPixelDataError error(*trailer, errorType, fedId);
+      errors[dummyDetId].push_back(error);
+    }
   }
   return fedTrailer.moreTrailers();
 }
 
-bool RPixErrorChecker::checkROC(bool& errorsInEvent, int fedId,  Word32& errorWord) const
+bool RPixErrorChecker::checkROC(bool& errorsInEvent, int fedId, uint32_t iD, Word32& errorWord,  Errors& errors) const
 {
   int errorType = (errorWord >> ROC_shift) & ERROR_mask;
   if likely(errorType<25) return true;
 
   switch (errorType) {
   case(25) : {
-    LogDebug("")<<"  invalid ROC=25 found (errorType=25)";
+    LogError("")<<"  invalid ROC=25 found (errorType=25)";
     errorsInEvent = true;
     break;
   }
   case(26) : {
-  //LogDebug("")<<"  gap word found (errorType=26)";
+    LogError("")<<"  gap word found (errorType=26)";
     return false;
```

Comment:
should this actually have a "return" here?

Reply:
It’s not a real error actually, just an empty word; that’s why it is treated differently.

Reply:
in this case LogError is not justifiable. Please demote to LogInfo or even LogDebug